src/mongo/db/audit/audit.cpp

```diff
@@ -278,7 +278,7 @@ class FileAuditLog : public WritableAuditLog {
         // first, then repeatedly write to that position if we
         // have to retry.
         auto pos = _file->tellp();
-        auto data = _membuf.str();
+        auto data = std::move(_membuf).str();
         _membuf.str({});
```

Comment:
Do we need `_membuf.str({})` after move?

Reply:
@igorsol ok, I can do it. I see two options here:
1) Add custom test in scons which would define dedicated flag indicating that the string is move-constructed
2) Add appropriate ifdef's for checking if the standard library is not affected by this bug. Looks like minimal required versions of gcc and msvc are fine. I just need to verify the correct combination for clang with libstdc++ and apple's clang version.

IMHO the 1st option is more reliable. What do you think?

Reply:
It looks that I confused two things. The mentioned bug is only about making an extra allocation in `basic_stringbuf::str() &&` overload but it still clears the buffer even without the fix. However, in clang versions < 17 this overload is not there and so the buffer is not cleared. This means that in order to decide if we need to explicitly clear the buffer or not we need to detect if this overload is provided. Either by checking for the standard library versions which contain this overload or by doing custom test in `SConstruct`. In some variants we might still have an extra allocation though.

Reply:
@igorsol I updated the PR with the custom test for scons. Could you please have another look?